### ISSUE TITLE
Credentials should not be stored in plain text in .ini file

### ISSUE THREAD

**Issue by timlinux** [ConservationInternational/trends.earth#341]
When reviewing the changes made by trends.earth to the QGIS ini file (e.g. ```~/.local/share/QGIS/QGIS3/profiles/Kartoza/QGIS```) we can find the API credentials stored in plain text:

![image](https://user-images.githubusercontent.com/178003/103630804-19886a00-4f3a-11eb-947a-56fc9a2d6beb.png)

It is best practice to use the QGIS authentication framework to do this. You can find a nice worked example of doing this in the Mergin plugin here:

https://github.com/lutraconsulting/qgis-mergin-plugin/blob/master/Mergin/utils.py#L54

**Comment by luipir** [comment]
seems not so hard to follow https://docs.qgis.org/3.16/en/docs/pyqgis_developer_cookbook/authentication.html#adapt-plugins-to-use-authentication-infrastructure (well I wrote is ;) ) using NAM instead of Requests.
BTW now lastest QGIS has some QNAM classes with blocking and not bloking capability (NAM was only blocking). I should check how to integrate.
@ricardogsilva The use of creadential management have to be evaluated because would change some authentication steps repect the original plugin. Not sure if it's the focus of the project.
IMHO it's an important aspect especially for plugins proposed by public institutions.

**Comment by azvoleff** [comment]
For sure - thanks for flagging, its been on the list to change this. Currently the credentials are only used to tie users to their submitted tasks on our API so as to get the URLs for the results of their tasks (the output of which are stored in a public bucket), and the passwords are random ones generated by our system, so it hasn't been a big security issue thus far.

But moving forward we've been looking towards using signed URLs to limit access to results, and better securing the credentials will be increasingly important. Once this is implemented I suppose we'll also need to request users to change their passwords given they were previously stored in plain-text, so we'll need to keep that in mind.
